server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -308,6 +321,7 @@ protected void discover() throws IOException {
                 throw new IOException("Could not read from extensions.yml", e);
             }
             for (Extension extension : extensions) {
+                extensionSettingsMap.put(extension.getUniqueId(), extension);
```

Comment:
loadExtension(...) has logic to ignore / error check that might cause the lists to be inconsistent, I'd recommend moving this `put` into the end of the load extension method

Reply:
I moved this so that its next to the call to `extensionIdMap.put` now.